### PR TITLE
fix(rule ticket): triggering action on actor update

### PR DESCRIPTION
**Pull request by Rom1-B**
When you have a rule whose criterion is the assignment of a group "XXX", the rule is triggered when you change the "assigned" field. If the value contains the group "XXX", the action is applied, even if this group was already present before the current update.

IMHO, the rule should only apply at the time of adding the group and not if you add more actors afterwards.

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | !27454

### CODE REVIEW

**Comment by cedric-anne** [comment]
See https://github.com/glpi-project/glpi/issues/2047

Honestly, I do not know how to handle this. As long as it is not possible to make a difference between previous and updated values in rules engine, we cannot satisfy everyone. But, I am pretty sure that introducing such a change in a bugfixes version will make a lot of people unhappy, as they probably defined rules that acts on both existing and new actors.

**Comment by cconard96** [comment]
It depends on how people expect the rules engine to work. If they treat it specifically where it only acts on changes, then it may make sense.

More often it seems it is expected to act on the final fields after the change is made.

Example:
If there is a rule with criteria for a specific category and an assigned actor, they want it to trigger when the actor is already assigned and they change the category, add the actor after the category is already set, or trigger if both are done at the same time.

**Comment by trasher** [comment]
I agree with @cedric-anne; changing this behaviour is a breaking change; it cannot be done in bugfixes version.

"the rule should only apply at the time of adding the group " => this may be true on some case, but not on others... A correct solution would require (much?) more work.

**Comment by Rom1-B** [comment]
In fact, the rule was executed even if the actors were not modified.
The last commit must correct it.

**Comment by cedric-anne** [comment]
As discussed IRL, it fixes a regression introduced in GLPI 10.0.x.
Please add a test case that validates the fixed behaviour.

File: src/Ticket.php [hunk 1]
```diff
@@ -1189,6 +1190,14 @@ public function prepareInputForUpdate($input)
                 $input_key = '_' . $field . '_' . $t;
                 $deleted_key = $input_key . '_deleted';
                 $deleted_actors = array_key_exists($deleted_key, $input) && is_array($input[$deleted_key]) ? array_column($input[$deleted_key], 'items_id') : [];
+                $tmp_input = $input[$input_key] ?? [];
+                if (!is_array($tmp_input)) {
+                    $tmp_input = [$input[$input_key]];
```
Comment (cedric-anne) [inline src/Ticket.php:1195]:
```suggestion
                    $tmp_input = [$tmp_input];
```

**Review by cedric-anne — APPROVED** [review]

**Review by orthagh — APPROVED** [review]

**Pull request merged**